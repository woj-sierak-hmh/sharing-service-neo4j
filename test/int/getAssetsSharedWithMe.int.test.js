const fetch = require('node-fetch');
//const { setupIntTest, teardownIntTest } = require('./utils');
const {intSetup, testRequest} = require('./utils');
const config = require('../../app/config.js');

const host = `${config.get('test:int:host')}:${config.get('test:int:port')}/`;

describe('getShares', () => {
  describe('As a teacher recipient, I...', () => {
    it('should be able to retrieve assets shared with me', async () => {
      // GET /v2/access/tenant/{tenantRefId}/user/{userRefId}/asset/{assetType}/assets
      const district = 'districtA';
      const recipient = 'teacherA2';
      const resourceType = 'PLAN';

      const query = `v2/access/tenant/${district}/` +
        `user/${recipient}` +
        `/asset/${resourceType}/assets`;

      const fetchResult = await testRequest({query, method: 'GET'});

      const parsedResult = await fetchResult.json();

      console.log('========================')
      console.log(parsedResult);
      console.log('========================')

      expect(fetchResult.status).toBe(200);

      expect(parsedResult).toMatchSnapshot({ 
        assets:[{
          assetRefId: 'plan2',
          assetType: 'PLAN',
          shareRefId: 'teacherA3B1',
          shareDate: expect.any(String) 
        },
        { 
          assetRefId: 'plan1',
          assetType: 'PLAN',
          shareRefId: 'teacherA1',
          shareDate: expect.any(String) 
        }]
      });

    });
  });
});