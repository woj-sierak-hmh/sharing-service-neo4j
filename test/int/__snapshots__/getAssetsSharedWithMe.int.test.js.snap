// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getShares As a teacher recipient, I... should be able to retrieve assets shared with me 1`] = `
Object {
  "assets": Array [
    Object {
      "assetRefId": "plan2",
      "assetType": "PLAN",
      "shareDate": Any<String>,
      "shareRefId": "teacherA3B1",
    },
    Object {
      "assetRefId": "plan1",
      "assetType": "PLAN",
      "shareDate": Any<String>,
      "shareRefId": "teacherA1",
    },
  ],
}
`;
